File: src/main/java/com/kingmang/lazurite/console/output/Output.java
package com.kingmang.lazurite.console.output;

import java.io.File;

public interface Output {

    String newline();

    void print(String value);

    default void print(Object value) {
        print(value.toString());
    }

    default void println() {
        print(newline());
    }

    default void println(String value) {
        print(value);
        println();
    }

    default void println(Object value) {
        println(value.toString());
    }

    default void error(Throwable throwable) {
        error(throwable.toString());
    }

    void error(CharSequence value);

    File fileInstance(String path);

}

File: src/main/java/com/kingmang/lazurite/console/output/Output.kt
package com.kingmang.lazurite.console.output

import java.io.File

interface Output {
    fun newline(): String

    fun print(value: String)

    fun print(value: Any?) =
        print(value.toString())

    fun println() =
        print(newline())

    fun println(value: String) {
        print(value)
        println()
    }

    fun println(value: Any?) =
        println(value.toString())

    fun error(throwable: Throwable) =
        error(throwable.toString())

    fun error(value: CharSequence)

    fun fileInstance(path: String): File
}
